### PR TITLE
HOCS-4268 - Adding user that already exists logs errors

### PR DESCRIPTION
**Pull request by agraham19**
* Warn instead of error on attempt to add duplicate user

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/info/domain/exception/RestResponseEntityExceptionHandler.java [hunk 1]
```diff
@@ -40,6 +40,12 @@ public ResponseEntity handle(ApplicationExceptions.EntityCreationException e) {
         return new ResponseEntity<>(e.getMessage(), INTERNAL_SERVER_ERROR);
     }
 
+    @ExceptionHandler(ApplicationExceptions.UserAlreadyExistsException.class)
+    public ResponseEntity handle(ApplicationExceptions.UserAlreadyExistsException e) {
+        log.warn("ApplicationExceptions.UserAlreadyExistsException: {}, Event: {}", e.getMessage(), value(EVENT, e.getEvent()));
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/domain/exception/RestResponseEntityExceptionHandler.java:45]:
```suggestion
        log.warn("ApplicationExceptions.UserAlreadyExistsException: {}", e.getMessage(), value(EVENT, e.getEvent()));
```
The `value(EVENT, e.getEvent())` is used by Kibana - IntelliJ just moans about it.

File: src/main/java/uk/gov/digital/ho/hocs/info/domain/exception/ApplicationExceptions.java [hunk 2]
```diff
@@ -124,6 +124,12 @@ public LogEvent getEvent() {
 
     }
 
+    class UserAlreadyExistsException extends EntityAlreadyExistsException {
+        public UserAlreadyExistsException() {
+            super("User with email already exists");
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/domain/exception/ApplicationExceptions.java:129]: This constructor leads to the event type being: `LogEvent.UNCAUGHT_EXCEPTION`, could you bubble through an appropriate event?

Comment (agraham19) [reply src/main/java/uk/gov/digital/ho/hocs/info/domain/exception/ApplicationExceptions.java]: I have made this change

File: src/main/java/uk/gov/digital/ho/hocs/info/domain/exception/RestResponseEntityExceptionHandler.java (hunk 1 above)
Comment (agraham19) [reply src/main/java/uk/gov/digital/ho/hocs/info/domain/exception/RestResponseEntityExceptionHandler.java]: I have made this change

**Review by LiamMacP — APPROVED** [review]

**Pull request merged**